### PR TITLE
Firelocks can't close if an holofan is placed on their turf

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Title, placing an holofan on the firedoor's turf will prevent it from closing by any means. Removing the holofan will allow the door to close.

## Why It's Good For The Game
More holofan uses, less firelocks pain

## Changelog
:cl:
qol: firelocks won't close if there is an atmos holofan on their turf.
/:cl:

### CODE REVIEW

**Comment by Jhyrachy** [comment]
You already know that I love you ♥

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -254,6 +256,14 @@
 			nextstate = null
 			close()
 
+/obj/machinery/door/firedoor/proc/check_holofan()
+	var/turf/loc = get_turf(src)
+	for(var/obj/structure/holosign/barrier/atmos/holo in loc.contents)
+		if(!holo)
+			continue
```
Comment (LemonInTheDark) [inline code/game/machinery/doors/firedoor.dm:263]: what? why would a holosign be on the turf, but also not exist?
also please don't call turfs loc in var names, just gets confusing
can't you do `return !!(locate(/obj/structure/holosign/barrier/atmos/) in turf_var)`?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Good idea, code is a pain tho.
I'm also a bit worried about the cost of checking for holo-barriers every time a firelock is dropped, maybe apply a trait to the turf in question when it gains a holo-barrier, and remove it when the barrier moves or is deleted?

**Comment by Ghilker** [comment]
> Good idea, code is a pain tho.
> I'm also a bit worried about the cost of checking for holo-barriers every time a firelock is dropped, maybe apply a trait to the turf in question when it gains a holo-barrier, and remove it when the barrier moves or is deleted?

i will make a trait for the turfs that is removed on holo barrier deletion

**Comment by Mothblocks** [comment]
A trait is okay, though I think the most modular approach here is to have a signal doors send when they're closing, and have holofans place an element that cancels it.

**Comment by Ghilker** [comment]
> A trait is okay, though I think the most modular approach here is to have a signal doors send when they're closing, and have holofans place an element that cancels it (and remove it when they go away).

i'll leave that to someone better than me lol

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -436,6 +436,8 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 /// Minor trait used for beakers, or beaker-ishes. [/obj/item/reagent_containers], to show that they've been used in a reagent grinder.
 #define TRAIT_MAY_CONTAIN_BLENDED_DUST "may_contain_blended_dust"
 
+#define TRAIT_FIREDOOR_STOP "firedoor_stop"
```
Comment (Mothblocks) [inline code/__DEFINES/traits.dm:439]: Doc-comment it.

**Comment by Mothblocks** [comment]
Making this an element is a mostly trivial exercise to learn elements, if you really want to.

**Comment by cacogen** [comment]
I swear to god this was based on my idea of being able to replace firelocks with holofirelocks

**Comment by Ghilker** [comment]
> I swear to god this was based on my idea of being able to replace firelocks with holofirelocks

No

**Comment by Jhyrachy** [comment]
Discord brainstorming gang

**Comment by cacogen** [comment]
I swear to god whoever suggested it stole my idea

**Comment by Jhyrachy** [comment]
> 
> 
> I swear to god whoever suggested it stole my idea

Sorry, I got the idea as a CE on the verge of a nervous breakdown

**Comment by cacogen** [comment]
It's a good idea this PR is nice

**Pull request merged**